lib/gc/Transforms/GPU/AllocsToSLM.cpp

```diff
@@ -62,22 +65,88 @@ struct ConvertAlloc : public OpRewritePattern<memref::AllocOp> {
                                          "Only support allocs in GPU regions");
     }
 
+    auto launchOp = allocOp->getParentOfType<gpu::LaunchOp>();
+
+    auto xSz = dyn_cast<arith::ConstantIndexOp>(
+        launchOp.getBlockSizeX().getDefiningOp());
+    auto ySz = dyn_cast<arith::ConstantIndexOp>(
+        launchOp.getBlockSizeY().getDefiningOp());
+    auto zSz = dyn_cast<arith::ConstantIndexOp>(
+        launchOp.getBlockSizeZ().getDefiningOp());
+
+    if (!xSz || !ySz || !zSz)
+      return rewriter.notifyMatchFailure(
+          allocOp, "Only support constant block sizes for now");
+
+    int64_t xI = xSz.value();
+    int64_t yI = ySz.value();
+    int64_t zI = zSz.value();
+
+    if (zI != 1)
+      return rewriter.notifyMatchFailure(
+          allocOp, "Only support 2D shared memory for now");
+
+    int64_t totalWorkGroupSize = xI * yI * zI;
```

Comment:
```suggestion
    int64_t totalWorkGroupSize = xI * yI;
```
zI is always 1 here.